BLSQ/iaso
# Polio 1291:  Vaccine Supply chain, details view front-end

Details view of polio vaccine supply chain

Related JIRA tickets : POLIO-1291

## Self proofreading checklist


## Changes

- Small improvement to rerouting to user/plugin homepage
- Add details page
    - each tab can be saved separately
    - option to save all
    - updates are managed through multiple PATCH, POST, DELETE requests, i.o 1 big PUT request like polio campaigns page 
- Add `MultiDateCell` and `MultiDateTimeCellRfc` for cleaner display of multiple dates in table cells
- Add `useSkipEffectUntilValue` hook, to block `useEffect` trigger before data is fetched
-  Fix some bugs in `TextArea` component (related to debounce feature)
- Add more generic `TabWithInfoIcon` component, derived from `PolioTab`
- Improve `useObjectState` hook (handle some edge cases)
- Add `useTab` hook, to handle `Tab` state and change handler
- removed `isTouched` function from utils. Replaced with more reliable `get` from lodash
- Add `shrinkLabel` prop to polio's `TextInput` to make UI look more consistent
- Remove vaccine tab related code in translations and validation
- update `useGetCampaigns` to enable passing options to the underlying `useQuery`hook


## How to test

*Setup*:
Enable `SHOW_DEV_FEATURES` feature flag on your local DB polio account

*Tests*:
- Go to VRF list. 
- Click create:
    - Pre-alerts and VAR tabs should be disabled
    - Fill in fields
    - click cancel: fields should be cleared
    - Fill in fields again and save:
        - Pre-alerts and VAR tabs should be active
        - The top bar title should be updated
        - There should be a "Save All" button
   - go to either PreAlerts or VAR:
       - click add button
       - clicking the delete before saving should remove the pre-alert/var from the UI
       - Go to VRF tab and change a field without saving
       - Go to prealert and save the prealert(s)
       - Go back to VRf tab: the change made before saving the prealert should not have overwritten
       - Click cancel: the change in the vrf should be reverted

Try variations of this flow with:
- Using save all when updating 1,2 or 3 tabs
- Saving one tab, then saving another, saving all, or canceling
- In pre-alerts and VAR: try adding, updating and deleting entries with a single save 

Note: 
- When saving multiple tabs, each tab should have its own snackbar if update is successful
- When saving, if there is an error, each call will have it's own error snackbar (create, update, delete)

## Print screen / video


https://github.com/BLSQ/iaso/assets/38907762/e55187bf-5611-4a46-8688-4bab244c63c9


## Notes

Merges into the backend branch https://github.com/BLSQ/iaso/tree/POLIO-1301-vaccine-supply-chain-backend-part-2

failing python tests will be addressed there
